contracts/ServiceNodeContribution.sol

```diff
@@ -76,7 +80,8 @@ contract ServiceNodeContribution is Shared {
         address _stakingRewardsContract,
         uint256 _maxContributors,
         BN256G1.G1Point memory _blsPubkey,
-        IServiceNodeRewards.ServiceNodeParams memory _serviceNodeParams
+        IServiceNodeRewards.ServiceNodeParams memory _serviceNodeParams,
+        IServiceNodeRewards.Contributor[] memory _reservedContributors
```

Comment:
There are two problems with having this in the constructor:
- it isn't updateable if the operator wants to reuse the contract
- acceptable reserved contributions *depend* on the amount the operator contributes (both for "does this meet the minimum" and "does this not exceed the staking requirement" checks), and the operator contribution isn't known here.

I think both can be solved by taking `_reservedContributors` as a parameter of `contributeOperatorFunds` (or perhaps having an alternative version called something like `contributeOperatorFundsWithReservations`), since that seems to be the earliest point where we know the operator amount and thus can validate the reservation amounts.